shell/platform/embedder/tests/embedder_unittests_proctable.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/embedder/embedder.h"
+
+#include "flutter/testing/testing.h"
+
+namespace flutter {
+namespace testing {
+
+// Verifies that the proc table is fully populated.
+TEST(EmbedderProcTable, AllPointersProvided) {
+  FlutterEngineProcTable procs = {};
+  procs.struct_size = sizeof(FlutterEngineProcTable);
+  FlutterEngineGetProcAddresses(&procs);
+
+  void (**proc)() = reinterpret_cast<void (**)()>(&procs.create_aot_data);
```

Comment:
Another way of doing this that doesn't depend on the fact that the first proc address is `create_aot_data` is to get the size of the struct, then negate the struct_size field from the same, then divide the remainder by the pointer size to get the number of procs in the table. Cycle from the first proc after the struct_size field offset and check for nullptr.

Reply:
Same technique can apply in the other tests. Your call in whether you want to use the suggestion though since we know this can't change now.

Reply:
I considered that, but that relies on the address of the first proc being equivalent to the struct start + `sizeof(struct_size)` which AFAIK isn't guaranteed due to struct alignment. (Doing the proc iteration itself seemed safe to me since alignment that isn't the size of a pointer seems completely implausible.) Since, as you mentioned, we are guaranteed never to re-order the struct, this version seemed safer.